### PR TITLE
feat(provider/azure): Use Azure Managed Disk instead of VHD when creating server group

### PR DESCRIPTION
**Pull request by michaeljqzq**
* Update azure server group resource template to use managed disks

* Fix the ImageReference structure in Azure resource template

* Update apiversion to support managed disk

* Fix test case

### CODE REVIEW

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/templates/AzureServerGroupResourceTemplate.groovy [hunk 1]
```diff
@@ -33,6 +33,7 @@ package com.netflix.spinnaker.clouddriver.azure.templates
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.databind.SerializationFeature
 import com.netflix.spinnaker.clouddriver.azure.resources.servergroup.model.AzureServerGroupDescription.AzureInboundPortConfig
+import com.netflix.spinnaker.clouddriver.model.Image
 import groovy.util.logging.Slf4j
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/templates/AzureServerGroupResourceTemplate.groovy]: Looks like the sort-order is off here.

Comment (michaeljqzq) [reply clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/templates/AzureServerGroupResourceTemplate.groovy]: I found that this import is actually not used in the code. Removed it

**Comment by mybayern1974** [comment]
@duftler , a kindly ping on this PR since it blocks the basic e2e flow of Azure.

**Comment by michaeljqzq** [comment]
Hi @duftler  @lwander  Could you help review this pull request since another pull request in rosco repository is already merged. Thanks

**Pull request merged**